src/main/kotlin/app/batch/DecompressionProcessor.kt

```diff
@@ -0,0 +1,36 @@
+package app.batch
+
+import org.apache.commons.compress.compressors.CompressorStreamFactory
+import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.batch.item.ItemProcessor
+import org.springframework.stereotype.Component
+import java.io.InputStream
+
+@Component
+class DecompressionProcessor : ItemProcessor<DecryptedStream, InputStream> {
+
+    override fun process(item: DecryptedStream): InputStream? {
+        val inputStream = item.inputStream
+        val fileName = item.fileName
+        try {
+            val compressorInputStream = CompressorStreamFactory().createCompressorInputStream(CompressorStreamFactory.GZIP,
+                inputStream) as GzipCompressorInputStream
+            logger.info("Compressed size of the file $fileName : ${compressorInputStream.compressedCount}")
+            return compressorInputStream
+        }
+        catch (e: Exception) {
+            val deCompressionExceptionMsg = "Exception occurred when decompressing the gzip decrypted input stream from the file $fileName"
+            logger.error(deCompressionExceptionMsg)
+            throw RuntimeException("$deCompressionExceptionMsg : $e")
```

Comment:
Do we want to send the whole Exception here? It might be of more use to send the message only.
Also should we define a more specific error than `RuntimeException` ?

Reply:
I believe it just prints the exception name and message. We may not need any specific exception in this case as the the idea is to  blow up  rather than handling it

Reply:
Here is the output from my log file - when the test is run - should this not contain an error signature? 
`[2019-10-08 11:31:09] [ERROR] [Test worker] [DecompressionProcessor.kt:process:25] Exception occurred when decompressing the gzip decrypted input stream from the file test.json.gz`